### DISCUSSION TITLE
Up navigation button doesn't work right

### DISCUSSION THREAD

**Discussion by camsteffen** [community/community#3916] [Mobile]
The up navigation button (left arrow in the top left corner) seems to just mimic the back button which is wrong and frustrating.

For example, when I click "view it on Github" in an email to open the app, the up button should take me to the main screen. Instead, it exits the app. This goes against [Principles of navigation](https://developer.android.com/guide/navigation/navigation-principles#the_up_button_never_exits_your_app).

> ### The Up button never exits your app
>
> If a user is at the app's start destination, then the Up button does not appear, because the Up button never exits the app. The Back button, however, is shown and does exit the app.
>
> When your app is launched using a deep link on another app's task, Up transitions users back to your app’s task and through a simulated back stack and not to the app that triggered the deep link. The Back button, however, does take you back to the other app.

**Comment by BudgieInWA** [comment]
> ... is wrong and frustrating.

Agreed! As I instinctively use the Up and Back buttons to move around the app, I am constantly thrown off, baffled and completely lost within the app. As an android user, the "up" vs "back" distinction is ingrained into my subconscious, and this app proves it to me by constantly breaking expectations that I didn't even know I was relying on so often.

Made worse by the fact that the "back" behaviour is also wrong when following a deep link: the back button takes you through the full history *within* the app, before exiting to take me back to whichever app showed me the deep link. When I open a gihub link, I have no idea how many times I have to press "back" to get back to where I was, because I'm taken through the history from whatever I was doing last time I was using the GutHub app.

The current behaviour results in a soft lock of the app, making it impossible to get to the main screen after following a deep link:

1. Close the app (via the app switcher, or restart the phone) to clear any history.
2. Follow a deep link into some discussion.
3. Use the Up button, which takes you to the previous app. (This is the expected behaviour for the Back button, not the Up button).
4. Use the app switcher to return to the GitHub app, which will still be on the discussion screen.
5. Use the Up button, which closes the app. The expectation is that it navigates up the hierarchy instead.
6. No matter how many times you return to the GitHub app via the app switcher the behaviour is the same.

The only way to get to the main screen is to close the app (via the app switcher), or to open the app from the app launcher after previously having closed the app via the up button (which is also unexpected).

The current navigation system might fit right in on iOS, but I am not qualified to evaluate that, because I am not an "iOS native". Ya'll need to consult/test with some "Android natives" and trust their muscle memory and subconscious instincts.

Android navigation needs to be rebuilt on the principal: **<kbd>⬅ Up</kbd> takes you up the heirarchy (like the breadcrumbs on a website), <kbd>◀ Back</kbd> takes you back in time (like the back button in the browser).**
